examples/features/load_balancing/client/main.go

```diff
@@ -123,4 +141,87 @@ func (*exampleResolver) Close()                                  {}
 
 func init() {
 	resolver.Register(&exampleResolverBuilder{})
+	balancer.Register(&customRoundRobinBuilder{})
+}
+
+const customRRName = "custom_round_robin"
+
+type customRRConfig struct {
+	serviceconfig.LoadBalancingConfig `json:"-"`
+
+	// N represents how often pick iterations chose the second SubConn
+	// in the list. Defaults to 3.
+	N uint32 `json:"n,omitempty"`
+}
+
+func (customRoundRobinBuilder) ParseConfig(s json.RawMessage) (serviceconfig.LoadBalancingConfig, error) {
+	lbConfig := &customRRConfig{
+		N: 2,
+	}
+	if err := json.Unmarshal(s, lbConfig); err != nil {
+		return nil, fmt.Errorf("custom-round-robin: unable to unmarshal customRRConfig: %v", err)
+	}
+	return lbConfig, nil
+}
+
+type customRoundRobinBuilder struct{}
+
+func (customRoundRobinBuilder) Name() string {
+	return customRRName
+}
+
+func (customRoundRobinBuilder) Build(cc balancer.ClientConn, bOpts balancer.BuildOptions) balancer.Balancer {
+	crr := &customRoundRobin{}
+	baseBuilder := base.NewBalancerBuilder(customRRName, crr, base.Config{HealthCheck: true})
+	crr.Balancer = baseBuilder.Build(cc, bOpts)
+	return crr
+}
+
+type customRoundRobin struct {
+	// Embeds balancer.Balancer because needs to intercept UpdateClientConnState
+	// to learn about N.
+	balancer.Balancer
+	n uint32
+}
+
+func (crr *customRoundRobin) UpdateClientConnState(state balancer.ClientConnState) error {
+	crrCfg, ok := state.BalancerConfig.(*customRRConfig)
+	if !ok {
+		return balancer.ErrBadResolverState
+	}
+	crr.n = crrCfg.N
+	return crr.Balancer.UpdateClientConnState(state)
+}
+
+func (crr *customRoundRobin) Build(info base.PickerBuildInfo) balancer.Picker {
+	if len(info.ReadySCs) == 0 {
```

Comment:
If you want to make this deterministic, then also do this until `len(info.ReadySCs) == 2`?

Reply:
Made deterministic in new PR based off child balancer connectivity states.